libs/application/template-api-modules/src/lib/modules/templates/children-residence-change/children-residence-change.service.ts

```diff
@@ -37,59 +37,62 @@ export class ChildrenResidenceChangeService {
 
   async submitApplication({ application }: props) {
     const { answers, externalData } = application
-    const { parentNationalRegistry } = externalData
-    const nationalRegistry = (externalData.nationalRegistry
-      .data as unknown) as User
+    const { nationalRegistry } = externalData
+    const applicant = nationalRegistry?.data
     const s3FileName = `children-residence-change/${application.id}.pdf`
     const file = await this.s3
       .getObject({ Bucket: this.presignedBucket, Key: s3FileName })
       .promise()
     const fileContent = file.Body as Buffer
 
-    const selectedChildren = application.externalData.childrenNationalRegistry.data.filter(
-      (c) => application.answers.selectChild.includes(c.name),
+    const selectedChildren = getSelectedChildrenFromExternalData(
+      applicant.children,
+      answers.selectChild,
     )
 
+    const otherParent = selectedChildren?.[0]?.otherParent
```

Comment:
Should we check array length and do we need to null check here?

Reply:
I don't think we need a check here, neither for length or the null check.